app/src/main/java/com/example/parkingsystem/mvp/model/ReservationModel.java

```diff
@@ -1,29 +1,81 @@
+    private static final String SECURITY_CODE = "Security code: ";
+    private static final String PLACE = "Place: ";
+    private static final String LINE_JUMP = "\n";
+
+    private ReservationInfomation reservationInfomation;
+    private String buttonText;
+    private Calendar startDateAndTime = null;
+    private Calendar finishDateAndTime = null;
+    private boolean startTime = false;
+    private boolean finishTime = false;
 
     @Override
-    public void saveDate(String date) {
-        this.date = date;
+    public void saveDate(int year, int month, int dayOfMonth) {
+        if (buttonText.equals(BUTTON_START_DATE)) {
+            startDateAndTime = Calendar.getInstance();
+            startDateAndTime.set(Calendar.DAY_OF_MONTH, dayOfMonth);
+            startDateAndTime.set(Calendar.MONTH, month - 1);
+            startDateAndTime.set(Calendar.YEAR, year);
+        } else {
+            finishDateAndTime = Calendar.getInstance();
+            finishDateAndTime.set(Calendar.DAY_OF_MONTH, dayOfMonth);
+            finishDateAndTime.set(Calendar.MONTH, month - 1);
+            finishDateAndTime.set(Calendar.YEAR, year);
+        }
     }
 
     @Override
-    public String getDate() {
-        return date;
+    public void saveTime(int hour, int minute) {
+        if (buttonText.equals(BUTTON_START_DATE)) {
+            startTime = true;
+            startDateAndTime.set(Calendar.HOUR_OF_DAY, hour);
+            startDateAndTime.set(Calendar.MINUTE, minute);
+        } else {
+            finishTime = true;
+            finishDateAndTime.set(Calendar.HOUR_OF_DAY, hour);
+            finishDateAndTime.set(Calendar.MINUTE, minute);
+        }
     }
 
     @Override
-    public void saveTime(String time) {
-        this.time = time;
+    public String saveReservation(String securityCode, String place) {
+        if (validFields(securityCode, place)) {
+            reservationInfomation = ReservationInfomation.getInstance();
+            Reservation reservation = new Reservation(startDateAndTime, finishDateAndTime, securityCode, place);
+            reservationInfomation.putReservation(reservation);
+            SimpleDateFormat simpleDateformat = new SimpleDateFormat(FORMAT_DATE);
+            String startDate = simpleDateformat.format(reservation.getStartDateAndTime().getTime());
+            String finishDate = simpleDateformat.format(reservation.getFinishDateAndTime().getTime());
+            return START_DATE + startDate + LINE_JUMP +
+                    FINISH_DATE + finishDate + LINE_JUMP +
+                    SECURITY_CODE + reservation.getSecurityCode() + LINE_JUMP +
+                    PLACE + reservation.getPlace();
+        }
+        return null;
+    }
+
+    private boolean validFields(String securityCode, String place) {
+        if (startDateAndTime != null && finishDateAndTime != null && startTime && finishTime && !securityCode.isEmpty() && !place.isEmpty())
```

Comment:
small improve could be to use an OR and check if only one is missing to return false.
the order its imoportant too, most common checks to be false, should be at the left. this decreese number of checks

Reply:
done